Apps/frontend/src/components/WaveControls.svelte

```diff
@@ -7,10 +7,48 @@
   import { timeSweep } from "../stores.js";
   import TimeSweepSlider from "./TimeSweepSlider.svelte";
   import ThicknessSwitch from "./ThicknessSwitch.svelte";
+  import DistributeOffsetButton from "./DistributeOffsetButton.svelte";
+  import startStopLine from "./Waves.svelte";
   export let waveElement;
   export let isEnabled;
   export let indicatorElement;
   let btnOnOff;
+
+  let activeChannels = [];
+  let activeChannelCounter = NUM_CHANNELS;
+
+  let distributed = false;
+
+  const setActiveChannels = () => {
+    for (let i = 0; i < NUM_CHANNELS; i++) {
+      activeChannels[i] = false;
+    }
+  };
+
+  setActiveChannels();
+
+  const countActiveChannels = () => {
+    let count = 0;
+    for (let i = 0; i < NUM_CHANNELS; i++) {
+      if (activeChannels[i]) count++;
+    }
+    activeChannelCounter = count;
+  };
+
+  const distributeChannels = () => {
```

Comment:
This is the job of the `DistributeOffsetButton`, I'd prefer it there.

Reply:
Yep me aswell, got moved